tests/pytest.org.md

```diff
@@ -0,0 +1,122 @@
+# Pytest organization proposal
+
+This proposal intends to achieve the following
+  - Have a standard way of categorizing tests
+  - Have some guidelines around test file organization
+  - Have a master wrapper for test execution
+  - Follow common documentation style
+  - Test result collection
+
+## Test categorization
+Leverage pytest custom markers to group tests based on topology, platform, features and connection type.
+
+```
+pytest.ini
+[pytest]
+markers:
+    topology(topo_name): The topologies this particular testcase can run against. topo_name can be individual topology names like 't0', 't1', 'ptf', 'any' or a comma separated like ('t0', 't1') if supported on multiple topologies
+    platform(platform_name): used for platform specific test(broadcom, mellanox, vs etc)
+    feature(feature_name): feature this test is written for. eg. acl, nat
+    connection_type(name): names can be 'fabric' (which indicates the presence of a fanout switch) or 'direct' if a testcase uses directly connected links
```

Comment:
So my understanding is that feature can be missing but is recommended to have.

But platform is optional unless limited?

How about connection_type? Do you see this as mandatory or optional only when needed to specify?

Should we make the statement?

Reply:
> So my understanding is that feature can be missing but is recommended to have.
NJ> Yes
> But platform is optional unless limited?
NJ> Yes. Should this get renamed to 'asic'?
> How about connection_type? Do you see this as mandatory or optional only when needed to specify?
NJ> It is optional. I don't think we will see this marker a lot
> Should we make the statement?
NJ> I will make this more explicit. Also I think we need another dimension - physical/vs. What do you think?

Reply:
Sure let's rename platform to ASIC to leave space to add hwsku in the future as community requested.

I think physical and vs can be added to asic?

Reply:
I would keep it separate from ASIC. I think it's a probable scenario that we'll have tests that can only be run on physical devices, irrespective of ASIC type.

Reply:
how about 'device_type' which can have values 'physical' and 'vs'? I will rename 'platform' to 'asic'

Reply:
That sounds good to me!